File: .sling/libs/sushi/eqela/jk-core/0.35.0/jk.script.lua
jk = jk or {}

jk.script = jk.script or {}

jk.script.Script = {}
jk.script.Script.__index = jk.script.Script
_vm:set_metatable(jk.script.Script, {})

function jk.script.Script._create()
	local v = _vm:set_metatable({}, jk.script.Script)
	return v
end

function jk.script.Script:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.script.Script'
	self['_isType.jk.script.Script'] = true
	self.entryPoints = nil
	self.ctx = nil
end

function jk.script.Script:_construct0()
	jk.script.Script._init(self)
	do _g.jk.fs.CurrentDirectory:set(_g.jk.env.CommonPath:getAppDirectory()) end
	self.ctx = _g.jk.console.ConsoleApplicationContext._construct0(_g.jk.console.ConsoleApplicationContext._create())
	return self
end

function jk.script.Script:print(message)
	_io:write_to_stdout(_g.jk.lang.String:asString(message) .. "\n")
end

function jk.script.Script:status(message)
	do _g.jk.log.Log:status(self.ctx, _g.jk.lang.String:asString(message)) end
end

function jk.script.Script:debug(message)
	do _g.jk.log.Log:debug(self.ctx, _g.jk.lang.String:asString(message)) end
end

function jk.script.Script:info(message)
	do _g.jk.log.Log:info(self.ctx, _g.jk.lang.String:asString(message)) end
end

function jk.script.Script:warning(message)
	do _g.jk.log.Log:warning(self.ctx, _g.jk.lang.String:asString(message)) end
end

function jk.script.Script:error(message)
	_vm:throw_error(_g.jk.lang.ExceptionWithString:forString(_g.jk.lang.String:asString(message), false))
end

function jk.script.Script:initialize()
end

function jk.script.Script:requireParameter(args, index, name)
	local v = nil
	if args ~= nil and index < #args then
		v = args[index + 1]
	end
	if _g.jk.lang.String:isEmpty(v) then
		do self:error("Missing parameter: <" .. _g.jk.lang.String:safeString(name) .. ">") end
	end
	do return v end
end

function jk.script.Script:executeFunction(script)
	if not (script ~= nil) then
		do return 0 end
	end
	do self:initialize() end
	do
		local _status, _err = _vm:execute_protected_call(function()
			do script() end
		end)
		if _err then
			local e = _err
			do _g.jk.log.Log:error(self.ctx, _g.jk.log.Log:asString(self.ctx, e)) end
			do return 1 end
		end
	end
	do return 0 end
end

function jk.script.Script:command(name, handler)
	local sname = _g.jk.lang.String:asString(name)
	if not (sname ~= nil) then
		do return end
	end
	if not (self.entryPoints ~= nil) then
		self.entryPoints = {}
	end
	self.entryPoints[sname] = handler
end

function jk.script.Script:main(args)
	do self:initialize() end
	do
		local cmd = nil
		if args ~= nil and #args > 0 then
			cmd = args[0 + 1]
		end
		if _g.jk.lang.String:isEmpty(cmd) then
			local sb = _g.jk.lang.StringBuilder._construct0(_g.jk.lang.StringBuilder._create())
			do sb:appendString("Missing command parameter: <") end
			do
				local first = true
				local array = _g.jk.lang.Map:getKeys(self.entryPoints)
				if array ~= nil then
					local n2 = 0
					local m = _g.jk.lang.Vector:getSize(array)
					do
						n2 = 0
						while n2 < m do
							local cc = array[n2 + 1]
							if cc ~= nil then
								if not first then
									do sb:appendCharacter(124) end
								end
								do sb:appendString(cc) end
								first = false
							end
							do n2 = n2 + 1 end
						end
					end
				end
				do sb:appendCharacter(62) end
				do _g.jk.log.Log:error(self.ctx, sb:toString()) end
				do return -1 end
			end
		end
		if self.entryPoints ~= nil then
			local handler = _g.jk.lang.Map:get(self.entryPoints, cmd)
			if handler ~= nil then
				local v = {}
				local nargs = #args
				do
					local n = 1
					while n < nargs do
						do _g.jk.lang.Vector:append(v, args[n + 1]) end
						do n = n + 1 end
					end
				end
				do
					local r = -1
					do
						local _status, _err = _vm:execute_protected_call(function()
							r = handler(v)
						end)
						if _err then
							local e = _err
							do _g.jk.log.Log:error(self.ctx, _g.jk.log.Log:asString(self.ctx, e)) end
							r = -1
						end
					end
					do return r end
				end
			end
		end
		do _g.jk.log.Log:error(self.ctx, "Unsupported command: `" .. _g.jk.lang.String:safeString(cmd) .. "'") end
		do return -1 end
	end
end

File: .sling/libs/web5/eqela/jk-core/0.36.0/jk.script.js
let JkScriptScript = function() {
	this.entryPoints = null;
	this.ctx = null;
};

JkScriptScript.prototype._t = { "JkScriptScript" : true };
JkScriptScript.prototype._tobj = JkScriptScript;

JkScriptScript.NEW = function() {
	var v = new JkScriptScript;
	return v.CTOR_JkScriptScript();
};

JkScriptScript.prototype.CTOR_JkScriptScript = function() {
	this.ctx = null;
	this.entryPoints = null;
	JkFsCurrentDirectory.set((JkEnvCommonPath.getAppDirectory()));
	this.ctx = JkConsoleConsoleApplicationContext.NEW();
	return this;
};

JkScriptScript.prototype.print = function(message) {
	console.log(JkLangString.asString(message));
};

JkScriptScript.prototype.status = function(message) {
	JkLogLog.status(this.ctx, (JkLangString.asString(message)));
};

JkScriptScript.prototype.debug = function(message) {
	JkLogLog.debug(this.ctx, (JkLangString.asString(message)));
};

JkScriptScript.prototype.info = function(message) {
	JkLogLog.info(this.ctx, (JkLangString.asString(message)));
};

JkScriptScript.prototype.warning = function(message) {
	JkLogLog.warning(this.ctx, (JkLangString.asString(message)));
};

JkScriptScript.prototype.error = function(message) {
	throw JkLangExceptionWithString.forString((JkLangString.asString(message)), false);
};

JkScriptScript.prototype.initialize = function() {
	;
};

JkScriptScript.prototype.requireParameter = function(args, index, name) {
	var v = null;
	if(args != null && index < args.length) {
		v = args[index];
	}
	if(JkLangString.isEmpty(v)) {
		this.error("Missing parameter: <" + (JkLangString.safeString(name)) + (">"));
	}
	return v;
};

JkScriptScript.prototype.executeFunction = function(script) {
	if(!(script != null)) {
		return 0;
	}
	this.initialize();
	try {
		script();
	}
	catch(e) {
		JkLogLog.error(this.ctx, (JkLogLog.asString(this.ctx, e)));
		return 1;
	}
	return 0;
};

JkScriptScript.prototype.command = function(name, handler) {
	var sname = JkLangString.asString(name);
	if(!(sname != null)) {
		return;
	}
	if(!(this.entryPoints != null)) {
		this.entryPoints = new Map;
	}
	this.entryPoints.set(sname, handler);
};

JkScriptScript.prototype.main = function(args) {
	this.initialize();
	var cmd = null;
	if(args != null && args.length > 0) {
		cmd = args[0];
	}
	if(JkLangString.isEmpty(cmd)) {
		var sb = JkLangStringBuilder.NEW();
		sb.appendString("Missing command parameter: <");
		var first = true;
		var array = JkLangMap.getKeys(this.entryPoints);
		if(array != null) {
			var n2 = 0;
			var m = array.length;
			for(n2 = 0 ; n2 < m ; n2++) {
				var cc = array[n2];
				if(cc != null) {
					if(!first) {
						sb.appendCharacter('|'.charCodeAt());
					}
					sb.appendString(cc);
					first = false;
				}
			}
		}
		sb.appendCharacter('>'.charCodeAt());
		JkLogLog.error(this.ctx, (sb.toString()));
		return -1;
	}
	if(this.entryPoints != null) {
		var handler = JkLangMap.get(this.entryPoints, cmd);
		if(handler != null) {
			var v = new Array;
			var nargs = args.length;
			for(var n = 1 ; n < nargs ; n++) {
				v.push(args[n]);
			}
			var r = -1;
			try {
				r = handler(v);
			}
			catch(e) {
				JkLogLog.error(this.ctx, (JkLogLog.asString(this.ctx, e)));
				r = -1;
			}
			return r;
		}
	}
	JkLogLog.error(this.ctx, "Unsupported command: `" + (JkLangString.safeString(cmd)) + ("'"));
	return -1;
};

JkScriptScript.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkScriptScript"] === true;
};
